packages/core/src/internal/proc.js

```diff
@@ -149,27 +148,7 @@ function createTaskIterator({ context, fn, args }) {
       )
 }
 
-export default function proc(
-  iterator,
-  stdChannel,
-  dispatch = noop,
-  getState = noop,
-  parentContext = {},
-  options = {},
-  parentEffectId = 0,
-  meta,
-  cont,
-) {
-  const { sagaMonitor, logger, onError, middleware } = options
-  const log = logger || _log
-
-  const logError = err => {
-    log('error', err)
-    if (err && err.sagaStack) {
-      log('error', err.sagaStack)
-    }
-  }
-
+export default function proc(iterator, parentContext, parentEffectId, meta, env, cont) {
```

Comment:
imho we should reorder those args

`env` & `iterator` should be first 2, im a little bit more inclined to put `env` as first as its constant and making it first it would be easier to curry `proc` (although not something we actually want to do)

`meta` should become very last as its the least important one and I think we should also check if its possible to put `parentEffectId` in the `meta` object too

OTOH `cont` is callback and it seems reasonable to put it last, maybe `cont` should be the one before last one (`meta`)?

let me know what do u think

Reply:
I think moving `env` to first is a good idea. I do not have opinions for the order of other arguments. 

Though very important, `proc()` is only called in two other places: one in `runForkEffect()`, and one in `runSaga()`. I will move `env` to first now, and we can refactor the order later on if necessary.

Reply:
why not use hash to imitate named arguments (instead of working with positional ones as we do now)?

Reply:
It's just a matter of flavor. I prefer positional arguments for most cases. And for options, I prefer object/hash.